### PR TITLE
add new handling of one-hot encoded groundtruth data

### PR DESCRIPTION
**Pull request by Bomme**
before it was always detected as multilabel-indicator, which might have lead to unexpected behaviour

### CODE REVIEW

File: src/shared.py [hunk 1]
```diff
@@ -40,19 +40,51 @@ def load_id2path(index_file):
...
+
+    Args:
+        y: numpy array with the groundtruth data
+
+    Returns:
+        target_type: string
+        Either "multiclass-indicator" or the result of
+        sklearn.utils.multiclass.type_of_target
+    """
+    scikit_learn_type = type_of_target(y)
```
Comment (palonso) [inline src/shared.py:59]: We always apply one-hot encoding to the labels (even to binary datasets) so I guess this will always return "multilabel-indicator" in our pipeline. We could turn the function into something more tailored to our particular case that returns `multi-class`,  `multi-label` or `not-supported` for example. As we don't need to support the rest of options in [type_of_target](https://scikit-learn.org/stable/modules/generated/sklearn.utils.multiclass.type_of_target.html).
we could also turn this line into   something like `assert type_of_target(y) ==  "multilabel-indicator"` more like a sanity check.

File: test/test_shared.py [hunk 2]
```diff
@@ -16,13 +16,22 @@ def test_auc():
...
-    # These values were computed using the previous implementation and match the
-    # results of scikit-learn when using default parameters
+    # These values were computed using default scikit-learn parameters
     np.testing.assert_allclose(roc_auc, 0.8611111)
-    np.testing.assert_allclose(pr_auc, 0.84444444)
+    np.testing.assert_allclose(pr_auc, 0.8444444)
+
+
+def test_type_of_groundtruth():
+    assert shared.type_of_groundtruth([1, -1, -1, 1]) == "binary"
```
Comment (palonso) [inline test/test_shared.py:31]: We transform binary datasets into multi-class, so this type of input shouldn't appear in our pipelines. Should we still test it?
Actually we may want to support binary models in the future.

File: src/shared.py (hunk 1 above)
Comment (Bomme) [reply src/shared.py:59]: My intention was to simply extend the `type_of_target` function with our specific issue that `multilabel-indicator` can also indicate our special case of `multiclass` data. 
If we want to check the type for allowed instances I would add another method for that. Or just do it in the method that only works for specific types.
Also, I see the advantage that the type that scikit learns sees is transparent to the user like this.

File: test/test_shared.py (hunk 2 above)
Comment (Bomme) [reply test/test_shared.py:31]: This is just to test the current semantics of the function that it will not interfere with other values of `type_of_target`. See my other comment.

File: src/shared.py (hunk 1 above)
Comment (Bomme) [reply src/shared.py:59]: On second thought, just raising a ValueError with the information on the type would also make it transparent...
I think finally I don't have a strong preference and it's your code base :)
Should I change it?

Comment (palonso) [reply src/shared.py:59]: Ok, I like your idea. Let's keep this simple and merge it as it is. If we need to test allowed types we can handle it in a new PR.
